src/types/EmbeddedPaymentElement.tsx

```diff
@@ -0,0 +1,427 @@
+   *  Defaults to true. If true, the customer can delete all saved payment methods.
+   *  If false, the customer can't delete if they only have one saved payment method remaining.
+   */
+  allowsRemovalOfLastSavedPaymentMethod?: boolean;
+  /**
+   * By default, EmbeddedPaymentElement will accept all supported cards by Stripe.
+   * You can specify card brands EmbeddedPaymentElement should block or allow payment for by providing an array of those card brands.
+   * Note: This is only a client-side solution.
+   * Note: Card brand filtering is not currently supported in Link.
+   */
+  cardBrandAcceptance?: PaymentSheetTypes.CardBrandAcceptance;
+  /** (Private Preview) This parameter is expected to be removed once we GA this feature
+   * When using customerSessions, allow users to update their saved cards
+   */
+  updatePaymentMethodEnabled?: boolean;
+  /** The view can display payment methods like “Card” that, when tapped, open a sheet where customers enter their payment method details.
+   * The sheet has a button at the bottom. `formSheetAction` controls the action the button performs.
+   */
+  formSheetAction?: EmbeddedFormSheetAction;
+  /** Controls whether the view displays mandate text at the bottom for payment methods that require it. If set to `false`, your integration must display `PaymentOptionDisplayData.mandateText` to the customer near your “Buy” button to comply with regulations.
+   * Note: This doesn't affect mandates displayed in the form sheet.
+   */
+  embeddedViewDisplaysMandateText?: boolean;
+  /**
+   * Determines the behavior when a row  is selected. Defaults to `.default`.
+   */
+  rowSelectionBehavior?: RowSelectionBehavior;
+}
+
+// -----------------------------------------------------------------------------
+// Embedded API
+// -----------------------------------------------------------------------------
+
+class EmbeddedPaymentElement {
+  /**
+   * Call this when the intent configuration changes (e.g., amount or currency).
+   * Cancels any in-progress update. Ensures the correct payment methods are shown and fields are collected.
+   * If the selected payment option becomes invalid, it may be cleared.
+   * Returns the final result of the update; earlier in-flight updates will return `{ status: 'canceled' }`.
+   */
+  async update(
+    intentConfig: PaymentSheetTypes.IntentConfiguration
+  ): Promise<EmbeddedPaymentElementUpdateResult> {
+    const result =
+      await NativeStripeEmbedded.updateEmbeddedPaymentElement(intentConfig);
+    return result;
+  }
+
+  /**
+   * Confirm the payment or setup intent.
+   * Waits for any in-progress `update()` call to finish before proceeding.
+   * May present authentication flows (e.g., 3DS) if required.
+   * Requires the most recent `update()` call to have succeeded.
+   * Returns the final result: success, failure, or cancellation.
+   */
+  async confirm(): Promise<EmbeddedPaymentElementResult> {
```

Comment:
Android doesn't return the result, but rather delivers it via a callback set at initialization time.

This is to support things like process death, as confirm might redirect to other applications.